engine_test.go

```diff
@@ -2209,6 +2209,63 @@ func TestReplaceIntoErrors(t *testing.T) {
 	}
 }
 
+func TestUpdate(t *testing.T) {
+	var updates = []struct {
+		updateQuery    string
+		expectedUpdate []sql.Row
+		selectQuery     string
+		expectedSelect  []sql.Row
+	}{
+		{
```

Comment:
Additional test cases you need:
* Setting the same column more than once. Copy MySQL's behavior for this
* Setting a non-existent column
* Setting null for a non-nullable column

Reply:
Also:
* Updates involving limit and offset, combined with order by
* Tests without where clauses (can be combined with above)
* Tests where not all matched rows were updates (you have a test where no rows were updated, but not one where some rows were updated and some were not)

Reply:
Also the verification queries aren't sufficient. Consider a hypothetical bug where the where clause was being ignored and every row updated. You need at least one tests that asserts the contents of an entire table after an update.